spring-context/src/main/java/org/springframework/cache/support/AbstractCacheManager.java

```diff
@@ -191,4 +196,11 @@ protected Cache getMissingCache(String name) {
 		return null;
 	}
 
+	public Set<CacheDecorator> getCacheDecorators() {
+		return cacheDecorators;
+	}
+
+	public void addCacheDecorator(CacheDecorator cacheDecorator) {
```

Comment:
That's better, but it's not really changing the base problem. If we want to introduce a general purpose cache decorator, then we need to provide a contract that allows a user to manipulate this properly while making sure the base class (the transaction aware one) can enforce the behaviour that its implementation brings.